src/level/platform.rs

```
@@ -0,0 +1,510 @@
+                    PlatformDirection::Forward => path[((curr_segment_index + 1) % path_len) as usize],
+                    PlatformDirection::Backward => path[(curr_segment_index - 1) as usize]
+                };
+                
+                let total_time = (PI * 8.0 * (previous_segment.x as f32 - next_segment.x as f32).abs()) / (2.0 * speed);
+                if curr_state == PlatformState::Play {
+                    platform.arc_time = platform.arc_time + time.delta_secs();
+                }
+                let curr_t = (platform.arc_time / total_time) * PI / 2.0;
+
+                let x_diff = next_segment.x - curr_segment.x;
+                let y_diff = curr_segment.y - previous_segment.y;
+                let other_y_diff = next_segment.y - curr_segment.y;
+                let other_x_diff = curr_segment.x - previous_segment.x;
+                match x_diff {
+                    x if x < 0 => match y_diff {
+                        x if x <= 0 => Vec2::new(-sin(curr_t), cos(curr_t)), // #5
+                        _ => Vec2::new(-sin(curr_t), -cos(curr_t)) // #2
+                    },
+                    x if x > 0 => match  y_diff {
+                        x if x <= 0 => Vec2::new(sin(curr_t), cos(curr_t)), // #6
+                        _ => Vec2::new(sin(curr_t), -cos(curr_t)) // #3
+                    },
+                    0 => match other_y_diff {
+                        x if x >= 0 => match other_x_diff {
+                            x if x >= 0 => Vec2::new(cos(curr_t), -sin(curr_t)), // #4
+                            _ => Vec2::new(-cos(curr_t), -sin(curr_t)) // 7
+                        },
+                        _ => match other_x_diff {
+                            x if x >= 0 => Vec2::new(cos(curr_t), sin(curr_t)), // #1
+                            _ => Vec2::new(-cos(curr_t), sin(curr_t)) // #8
+                        }
+                    },
+                    _ => Vec2::new(0.0, 0.0) // unreachable
```

If this is unreachable, consider annotating it with the `unreachable!()` macro so that the function doesn't silently fail.